internal/service/componentdescriptor/gitsources.go

```diff
@@ -52,7 +52,11 @@ func (s *GitSourcesService) AddGitSources(componentDescriptor *compdesc.Componen
 	if err != nil {
 		return fmt.Errorf("failed to get latest commit: %w", err)
 	}
-	access := github.New(gitRepoURL, "", latestCommit)
+	commitOption := func(spec *github.AccessSpec) {
+		spec.Commit = latestCommit
+	}
+
+	access := github.New(gitRepoURL, "", commitOption)
```

Comment:
I think this may be simplified. The `github` package already provides a [`WithCommit` func](https://github.com/open-component-model/ocm/blob/main/api/ocm/extensions/accessmethods/github/method.go#L94-L98):

```go
access := github.New(gitRepoURL, "", github.WithCommit(latestCommit))
```